### ISSUE TITLE
user object warning logged, even when not touching `session.user`

### ISSUE THREAD

**Issue by j4w8n** [supabase/supabase-js#1709]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

With supabase-js@2.42.5 and the ssr auth helper package, after a user logs in, the below warning is logged to the server console five times, with a fairly minimal SvelteKit app. This happens despite the fact that none of my code is calling `session.user`, nor am I destructuring the user property from `session`, or destructuring with `...rest`-type syntax.

_Using the user object as returned from supabase.auth.getSession() or from some supabase.auth.onAuthStateChange() events could be insecure! This value comes directly from the storage medium (usually cookies on the server) and many not be authentic. Use supabase.auth.getUser() instead which authenticates the data by contacting the Supabase Auth server._

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

https://github.com/j4w8n/getsession-warning

## Expected behavior

The warning should not log when dev code doesn't explicitly call `session.user`.

## Root Cause

I figured out why the first three logs happen: the Supabase client's server-side storage, from +layout.ts, is returning stringified JSON. You can see this in the [docs](https://supabase.com/docs/guides/auth/server-side/creating-a-client?queryGroups=framework&framework=sveltekit&queryGroups=environment&environment=layout#creating-a-client). `JSON.stringify()` causes each property of `session` to be touched during the stringify process, which inevitably calls `session.user`. So, each time `_useSession()` is called, whether implicitly through dev code like `getSession()` or explicitly with internal auth-js processes, the warning is being logged.

I can't figure out where the final two warning logs are coming from. Could be from the same client initialization as above, but perhaps whatever triggers it is getting run asynchronously, so it's logging later.

Even if this warning wasn't an issue in the context of ssr, it could still be problematic, as the auth-js client itself calls `session.user` when updating a user, setting a session, refreshing a session, and more - at least some of which could theoretically happen server-side.

## System information

- OS: Windows, with or without WSL.
- Browser N/A
- Version of supabase-js: 2.42.5
- Version of ssr: 0.3.0, but it should happen on earlier versions too I'd think.
- Version of Node.js: N/A

## Additional context

https://github.com/supabase/auth-js/issues/873
https://github.com/supabase/auth-js/issues/873#issuecomment-2067684465
https://tc39.es/ecma262/multipage/structured-data.html#sec-json.stringify
https://tc39.es/ecma262/multipage/structured-data.html#sec-serializejsonproperty
https://github.com/supabase/auth-js/blob/master/src/GoTrueClient.ts#L1111-L1124

**Comment by ICTVTS3** [comment]
I cannot believe I'm still periodically looking for new updates on this..

**Comment by samuel-cappelli** [comment]
Hello! Any updates about this issue?

**Comment by jdgamble555** [comment]
Does asymmetric keys fix this? They have not updated the SvelteKit example, so not sure yet?

https://supabase.com/blog/jwt-signing-keys

https://supabase.com/docs/guides/getting-started/tutorials/with-sveltekit

J

**Comment by rossvold** [comment]
Just tried to setup a test sveltekit project, following the documentation, and for the life of me couldn't figure out what i did wrong [here](https://supabase.com/docs/guides/auth/server-side/sveltekit).
Shocked to figure out that this issue is almost 2 years old... This is in no way, shape or form ready for production at all because of this issue.

Atleast inform sveltekit users at this page https://supabase.com/docs/guides/auth/server-side/sveltekit that this issue is there?

**Comment by jdgamble555** [comment]
I got mine working doing this:
```ts
import {
    PUBLIC_SUPABASE_ANON_KEY,
    PUBLIC_SUPABASE_URL,
} from "$env/static/public";
import { verifyAdminJWT } from "$features/admin/db/jwt";
import type { Database } from "$lib/database.types";
import { createServerClient } from "@supabase/ssr";
import type { User } from "@supabase/supabase-js";
import type { Handle } from "@sveltejs/kit";

export const handle: Handle = async ({ event, resolve }) => {

    event.locals.supabase = createServerClient<Database>(
        PUBLIC_SUPABASE_URL,
        PUBLIC_SUPABASE_ANON_KEY,
        {
            cookies: {
                getAll: () => event.cookies.getAll(),
                setAll: (cookiesToSet) => {
                    cookiesToSet.forEach(({ name, value, options }) => {
                        event.cookies.set(name, value, {
                            ...options,
                            path: "/"
                        });
                    });
                }
            }
        }
    );

    event.locals.safeGetSession = async () => {

        const {
            data: claimsData,
            error: claimsError
        } = await event.locals.supabase.auth.getClaims();

        if (claimsError) {
            console.error(claimsError);
            return {
                user: null
            };
        }

        if (!claimsData) {
            return {
                user: null
            };
        }

        const customClaims = claimsData.claims?.claims || null;

        const claims = claimsData.claims;

        const user = {
            ...claims,
            id: claims.sub
        } as unknown as User;

        return {  user  };
    };

    return resolve(event, {
        filterSerializedResponseHeaders(name) {
            return name === "content-range" ||
                name === "x-supabase-api-version";
        }
    });
};
```
A few issued noted:
1.  The types do not match up exactly, as `getClaims()` needs actual typing instead of `User` for the data type besides JWT type.
2.  This does NOT fix where `getSession()` is still called under the hood with functions like `getUser()`, `getAuthenticatorAssuranceLevel()`, and `onAuthStateChange()`, which is what I have been screaming above.

So you still need to add this:
```ts
event.locals.safeGetSession = async () => {
    /** @ts-expect-error: suppressGetSessionWarning is not officially supported */
    event.locals.supabase.auth.suppressGetSessionWarning = true;
    ...
```

@kangmingtay 

J

**Comment by jdgamble555** [comment]
@mandarini - Why was the transferred from `auth-js`? 

The code that needs to be fixed is not `supabase-js` but `auth-js`...

https://github.com/supabase/supabase-js/issues/1709#issuecomment-3376175717

J

**Comment by mandarini** [comment]
@jdgamble555 please take a look here: https://github.com/orgs/supabase/discussions/39197

**Comment by rossvold** [comment]
> I got mine working doing this:
> 
> import {
>     PUBLIC_SUPABASE_ANON_KEY,
>     PUBLIC_SUPABASE_URL,
> } from "$env/static/public";
> import { verifyAdminJWT } from "$features/admin/db/jwt";
> import type { Database } from "$lib/database.types";
> import { createServerClient } from "@supabase/ssr";
> import type { User } from "@supabase/supabase-js";
> import type { Handle } from "@sveltejs/kit";
> 
> export const handle: Handle = async ({ event, resolve }) => {
> 
>     event.locals.supabase = createServerClient<Database>(
>         PUBLIC_SUPABASE_URL,
>         PUBLIC_SUPABASE_ANON_KEY,
>         {
>             cookies: {
>                 getAll: () => event.cookies.getAll(),
>                 setAll: (cookiesToSet) => {
>                     cookiesToSet.forEach(({ name, value, options }) => {
>                         event.cookies.set(name, value, {
>                             ...options,
>                             path: "/"
>                         });
>                     });
>                 }
>             }
>         }
>     );
> 
>     event.locals.safeGetSession = async () => {
> 
>         const {
>             data: claimsData,
>             error: claimsError
>         } = await event.locals.supabase.auth.getClaims();
> 
>         if (claimsError) {
>             console.error(claimsError);
>             return {
>                 user: null
>             };
>         }
> 
>         if (!claimsData) {
>             return {
>                 user: null
>             };
>         }
> 
>         const customClaims = claimsData.claims?.claims || null;
> 
>         const claims = claimsData.claims;
> 
>         const user = {
>             ...claims,
>             id: claims.sub
>         } as unknown as User;
> 
>         return {  user  };
>     };
> 
>     return resolve(event, {
>         filterSerializedResponseHeaders(name) {
>             return name === "content-range" ||
>                 name === "x-supabase-api-version";
>         }
>     });
> };
> 
> A few issued noted:
> 
>     1. The types do not match up exactly, as `getClaims()` needs actual typing instead of `User` for the data type besides JWT type.
> 
>     2. This does NOT fix where `getSession()` is still called under the hood with functions like `getUser()`, `getAuthenticatorAssuranceLevel()`, and `onAuthStateChange()`, which is what I have been screaming above.
> 
> 
> So you still need to add this:
> 
> event.locals.safeGetSession = async () => {
>     /** @ts-expect-error: suppressGetSessionWarning is not officially supported */
>     event.locals.supabase.auth.suppressGetSessionWarning = true;
>     ...
> 
> [@kangmingtay](https://github.com/kangmingtay)
> 
> J

Wasn't able to implement this approach, still getting the warnings.

Wish this was taken more seriously by the Supabase team. Atleast this pushed me to learn how to implement my own auth...

**Comment by robinhayward** [comment]
Just starting out building a big SvelteKit app, took a while to decide on Supabase for the backend. I have lost time chasing this warning (error?) to eventually wind up here.. to find that nobody at Supabase cares. This is auth. Immediately aborting Supabase just based on this alone, cannot justify building out on a platform with an issue like this on day 1, that nobody cares about fixing. Disappointed.

**Comment by hf** [comment]
> Just starting out building a big SvelteKit app, took a while to decide on Supabase for the backend. I have lost time chasing this warning (error?) to eventually wind up here.. to find that nobody at Supabase cares. This is auth. Immediately aborting Supabase just based on this alone, cannot justify building out on a platform with an issue like this on day 1, that nobody cares about fixing. Disappointed.

I'm sorry you feel that way. I know the warning is annoying, but it's there for a very good reason. We prioritize security over everything else, and if you accidentally rely on the user object in server side rendering you might be making your application completely insecure. This is because anyone can pretend to be a real browser, and send whatever cookies it wants. Sadly the user object is stored in the cookies, and then it ends up directly into `getSession()`. So if you rely on that user object for anything of importance, I can spoof literally any user.

This is what the warning is yelling about.

`getClaims()` does resolve the security issue, by verifying the JWT. But the user object is still not usable unless you explicitly call `getUser()` which fetches it fresh from the server. 

We are unlikely to remove the warning all together, but will try to find another approach that has less false positives than the existing one.

**Comment by j4crev** [comment]
The biggest reason I opened this issue is because the Supabase sdk itself is triggering the warning in certain situations.  That’s what really needs addressed.

It sucks that SvelteKit has behavior that also triggers the warning, but that should be resolved once remote functions and async Svelte become more popular.

**Comment by jdgamble555** [comment]
@hf 

> getClaims() does resolve the security issue, by verifying the JWT. But the user object is still not usable unless you explicitly call getUser() which fetches it fresh from the server.

Please actually look at the problem at hand instead of citing it is just a "security issue." I have stated a clear solution several times in this thread.

`getClaims()` actually still calls `getSession()` under the hood [here](https://github.com/supabase/supabase-js/blob/086112e9ce0b3de6b22045bb2e1b43739dd6879e/packages/core/auth-js/src/GoTrueClient.ts#L3593).  


It is also IMPOSSIBLE to avoid calling `getSession()`, since the code STILL calls it, even if you don't want to. `session.user`, which is a proxy for `getSession()` is called 19 times in [GoTrueClient.ts](https://github.com/supabase/supabase-js/blob/086112e9ce0b3de6b22045bb2e1b43739dd6879e/packages/core/auth-js/src/GoTrueClient.ts#L1753), including in these situations:
- updateUser
- getUser
- refreshSession
- signOut
- linkIdentity
- unlinkIdentity
- linkIdentiyIdToken
- etc, etc

I think literally ALL user functions, including `getClaims()` will use the `session.user` proxy, which in itself calls `getSession()`, or call it directly. It is IMPOSSIBLE to avoid the warning.

No one is saying the security issue is not a threat, but having internal code call `getSession()`, even when your code doesn't use it directly, then stating the warning anyway, is extremely bad faith for the developers. Especially considering there have been hundreds of complaints in the issues alone, and it seems to be thrown to the back of the priority list with no hopeful fix in sight.

Two things need to happen to fix this issue:
1. Do NOT display the warning for when `getSession()` is called under the hood, only when `getSession()` is called directly in bad faith. The warning is for the developer, not for when it is called outside of the developer's control. The code is written poorly.
2. `getClaims()` is supposed to replace `getUser()`, but the typing it returns is terrible, and there has NOT been any updates to the docs on how to use it properly in the frameworks. Update ALL docs, and fix the return typing so that you can actually get the user information and use it without hacking together types.
3. When something like this is implemented, there should be a LOT of unit tests to show when it should and shouldn't be displayed. Again, the warning is displayed anyway, not when you say it is. Clearly, this was written without thinking about the side effects and testing them, and then it just gets cited (oh, the developers are the ones making the mistakes)... let's fix the problem instead of ignoring it please! More tests when something as crucial as this is implemented!

J

**Comment by rossvold** [comment]
> > Just starting out building a big SvelteKit app, took a while to decide on Supabase for the backend. I have lost time chasing this warning (error?) to eventually wind up here.. to find that nobody at Supabase cares. This is auth. Immediately aborting Supabase just based on this alone, cannot justify building out on a platform with an issue like this on day 1, that nobody cares about fixing. Disappointed.
> 
> I'm sorry you feel that way. I know the warning is annoying, but it's there for a very good reason. We prioritize security over everything else, and if you accidentally rely on the user object in server side rendering you might be making your application completely insecure. This is because anyone can pretend to be a real browser, and send whatever cookies it wants. Sadly the user object is stored in the cookies, and then it ends up directly into `getSession()`. So if you rely on that user object for anything of importance, I can spoof literally any user.
> 
> This is what the warning is yelling about.
> 
> `getClaims()` does resolve the security issue, by verifying the JWT. But the user object is still not usable unless you explicitly call `getUser()` which fetches it fresh from the server. 
> 
> We are unlikely to remove the warning all together, but will try to find another approach that has less false positives than the existing one. 

This low effort, uninformed response by supabase employees tells me everything i need to know about ever using Supabase again in the future.

**Comment by hf** [comment]
Thanks both for the feedback, I'll see what can be done ASAP.

**Comment by hf** [comment]
> getClaims() is supposed to replace getUser(), but the typing it returns is terrible, and there has NOT been any updates to the docs on how to use it properly in the frameworks. Update ALL docs, and fix the return typing so that you can actually get the user information and use it without hacking together types.

Just want to address this, as I'm sure someone will stumble upon it. `getClaims()` can't have much stronger types than it has today, as it extracts the JWT payload. This payload while somewhat standardized is often customized by developers themselves via our [Custom Access Token Hook](https://supabase.com/docs/guides/auth/auth-hooks/custom-access-token-hook). So you can put in or remove a lot of information in the JWT, and lots decide on this approach to improve performance of their apps.

Sadly it means that we can't auto-generate types, nor expose a stricter and friendlier interface for the function's return values. You're free to cast or wrap into a helper function that will return the proper types based on your project's hook.

**Comment by vehm** [comment]
I'm glad to see that this is finally getting attention. I still have a lot of love for Supabase, and truly believe it's a good platform to build from. With that being said, over a year of silence on this issue coupled with several frustrating limitations and feature gaps have made me consider steering the company I work for in another direction.

Not trying to air out personal grievances, just trying to stress how frustrating this sort of thing can be. The fact this issue has a PR opened mere hours after somebody finally got a pairs of eyes on it makes me very happy, but also leaves a sour taste in my mouth considering it's been over a year since this issue opened.

@j4w8n and @jdgamble555 (as well as others) have been invaluable in the interim, providing both temporary solutions and suggestions for a long-term fix. Thank you all.

**Comment by jdgamble555** [comment]
@hf 

> Just want to address this, as I'm sure someone will stumble upon it. getClaims() can't have much stronger types than it has today, as it extracts the JWT payload. This payload while somewhat standardized is often customized by developers themselves via our [Custom Access Token Hook](https://supabase.com/docs/guides/auth/auth-hooks/custom-access-token-hook). So you can put in or remove a lot of information in the JWT, and lots decide on this approach to improve performance of their apps.

> Sadly it means that we can't auto-generate types, nor expose a stricter and friendlier interface for the function's return values. You're free to cast or wrap into a helper function that will return the proper types based on your project's hook.

Yes, it is customizable, but Supabase added ALL of the fields itself. Yes, technically I can customize all of it, but if I wanted to veer away from Supabase, I would use my own typing.
```ts
{
  claims: {
    aal: 'aal1',
    amr: [ [Object] ],
    app_metadata: { provider: 'email', providers: [Array] },
    aud: 'authenticated',
    claims: {
      // ONLY HERE IS MY CUSTOM CLAIMS
    },
    email: 'jdgamble555@gmail.com',
    exp: 1761785325,
    iat: 1761781725,
    is_anonymous: false,
    iss: 'http://127.0.0.1:54321/auth/v1',
    phone: '',
    role: 'authenticated',
    session_id: 'f72e4c1a-e0a4-4f45-a8ae-c33f379268a3',
    sub: 'e0d4cf73-d53f-4d73-a5af-0d33dd242100',
    user_metadata: {
      avatar_url: ',
      email: 'jdgamble555@gmail.com',
      email_verified: true,
      full_name: 'Jonathan Gamble',
      iss: 'https://accounts.google.com',
      name: 'Jonathan Gamble',
      phone_verified: false,
      picture: '.',
      provider_id: '102863252967823414276',
      sub: '102863252967823414276'
    }
  },
  header: { alg: 'HS256', typ: 'JWT' },
  signature: Uint8Array(32) [
    ...
  ]
}
```
That is like saying, hey, technically you can customize postgres to be whatever you want, so we are not going to give you typing for the standard supabase functions... you can use your own GO functions etc...

This is a HUGE issue if basic typing coping the clear supabase pattern can't be used. It is close to `User`, but not quite due to `sub` and `id`, which BTW we CAN'T change how supabase handles this.

So yes, allow a standard type for this, and maybe an `overrideTypes()` function or something similar if we need to customize.

You guys have got to remember what makes supabase great is the DEVELOPER EXPERIENCE... without that, it is just another database.

J

**Comment by mandarini** [comment]
You’re all totally right to be annoyed about this, especially those of you on SvelteKit who kept saying “I’m not even touching session.user and it still logs.” That was a fair complaint.

The change I opened yesterday switches the warning to wrap only the user object on the server, not the entire session. So framework-level JSON/stringify/POJO checks on the session itself should stop triggering it. It also special-cases the common symbols so SvelteKit / Node inspection doesn’t spam warnings. The PR is under review right now.

We’ve been doing some repo/packaging restructuring as you may have noticed, which is part of why this didn’t get addressed sooner. This is not an excuse, just context. But that doesn’t make the experience you had any less frustrating. You were right to push.

And just to say it out loud: one bad GitHub issue shouldn’t condemn the whole tool. This is an auth-specific DX papercut, not “Supabase doesn’t care about security.” We do, that’s literally why the warning exists. Now we’re just making it fire when it actually matters. There are real people on this side who do care about DX, and this thread is actually useful for us because it shows where the friction is loudest.

Thanks for sticking with it a bit longer while we land this.

**Comment by jdgamble555** [comment]
Ok, what about `getClaims()` lack of documentation and types? Should I open another issues?

J

**Comment by mandarini** [comment]
Yes please @jdgamble555 . Thank you for this. Do so [here](https://github.com/supabase/supabase-js/issues) and choose "Documentation".

**Comment by mandarini** [comment]
Btw, you can try it out now in [v2.77.1-canary.1](https://github.com/supabase/supabase-js/releases/tag/v2.77.1-canary.1).

**Comment by jdgamble555** [comment]
@mandarini - What about updating the Types. Saying that it is not necessary because it is a customizable JWT is agreeing that Supabase doesn't care about DX again.

There is a clear pattern of 20+ Types that would need to be manually created, plus the `user_metadata`, `app_metadata`, etc. This NOT a `documentation` issue, but an internal coding issue. This could easily be put in the source code and allow a function to customize the types if necessary.

J

**Comment by jdgamble555** [comment]
NM, clear issue for it here - https://github.com/supabase/supabase-js/issues/1584

**Comment by mandarini** [comment]
All good, I know this stuff comes from wanting Supabase to be the best experience possible, and we share that goal.
This specific issue has definitely taken longer than anyone wanted, but it’s one piece of a much bigger picture. We’re actively improving both DX and security across the SDKs.

The PR here already helps a lot, and the typing discussion in supabase/supabase-js#1584 is exactly where we can keep improving things in parallel. Thanks again for the detailed feedback and for testing the canary. It really helps us tighten things up.

**Comment by jdgamble555** [comment]
@mandarini - Thanks for looking at this and getting this done quickly! I will test the canary version later this evening.

Here is the new `getClaims()` documentation issue for anyone following:

https://github.com/supabase/supabase/issues/40985

And again the `getClaims()` typing issue:

https://github.com/supabase/supabase-js/issues/1584

J

**Comment by jdgamble555** [comment]
@mandarini - So far so good! Will let you know here if I run into anything!

FWI - I only use Supabase on the server, so there could be things I missed.

Hopefully others test it!

Thank you!

J

**Comment by benoit-fuentes** [comment]
Hi! I'm not sure I fully understand the outcome of this thread. The issue has been closed but I'm still getting the warnings with the "Build a User Management App with SvelteKit" demo when using the lastest version (2.79.0) of supabase-js. Can someone summarize the situation?

**Comment by mandarini** [comment]
@benoit-fuentes the issue should have been fixed in this PR: https://github.com/supabase/supabase-js/pull/1817

You are saying that you are still experiencing the same issue? Can you please share your reproduction repository?

**Comment by benoit-fuentes** [comment]
@mandarini here is my code : https://github.com/benoit-fuentes/simple_app

It is not exactly the one suggested in the [tutorial](https://supabase.com/docs/guides/getting-started/tutorials/with-sveltekit?queryGroups=database-method&database-method=dashboard), since I simplified it even more (no more 'profile' table, you can just login with you mail and it shows "you're logged as [your mail]" when you do).

Warning is showed when I run the app locally in dev mode (cf photo above).

<img width="1403" height="857" alt="Image" src="https://github.com/user-attachments/assets/a3b5a6b2-b12c-4f7d-baa6-56d32bc0bf12" />

Note that I am a newbie in web development :)

**Comment by rossvold** [comment]
Setup a brand new project,
followed the setup guide exactly as instructed in your own docs.

Still happens both locally and using Vercel.

<img width="1000" height="450" alt="Image" src="https://github.com/user-attachments/assets/f5739039-4632-4e5e-88d4-09c6747587bd" />

**Comment by mandarini** [comment]
Ok, noted. We will look into it.

**Comment by rossvold** [comment]
> Ok, noted. We will look into it.

Here's the docs i referred to btw:
https://supabase.com/docs/guides/auth/server-side/sveltekit

**Comment by jdgamble555** [comment]
@rossvold - I believe the error actually is fixed, but both examples still use `getSession()`, which we no longer need to use.

https://supabase.com/docs/guides/getting-started/tutorials/with-sveltekit
https://supabase.com/docs/guides/auth/server-side/sveltekit

Here is my SvelteKit repo copying the example `User Management` app. I didn't get any issues, but let me know if you do:
https://github.com/jdgamble555/sveltekit-supabase-demo

I'm working to help revamp the docs for all Frameworks:

Follow the Docs Thread here:
https://github.com/supabase/supabase/issues/40985#issuecomment-3495846521

J

**Comment by rossvold** [comment]
> @rossvold - I believe the error actually is fixed, but both examples still use `getSession()`, which we no longer need to use.
> 
> https://supabase.com/docs/guides/getting-started/tutorials/with-sveltekit
> https://supabase.com/docs/guides/auth/server-side/sveltekit
> 
> Here is my SvelteKit repo copying the example `User Management` app. I didn't get any issues, but let me know if you do:
> https://github.com/jdgamble555/sveltekit-supabase-demo
> 
> I'm working to help revamp the docs for all Frameworks:
> 
> Follow the Docs Thread here:
> https://github.com/supabase/supabase/issues/40985#issuecomment-3495846521
> 
> J

I understand but this issue is not closed until the documentation is corrected then.

I'm not going to NOT follow the documentation and do something undocumented to get it working.

**Comment by mandarini** [comment]
@rossvold I undestand your very valid frustration. Please bear with us until we update the documentation. I will not close this issue until we update it, ok?

**Comment by jdgamble555** [comment]
@mandarini - Is this still being worked on? It's crazy to me the docs have been incorrect for this amount of time.

J

**Comment by mandarini** [comment]
@jdgamble555 I think there's another issue for that, including @ChrisChinchilla ? I have asked internally for this, please bear with us.

**Comment by chanmathew** [comment]
@mandarini any update on this? Every day this clutters our SSR logs and it's unbelievably frustrating how a simple thing like this has still not been resolved for over a year. I'm honestly about to migrate our paid account off Supabase because of this, not the mention the recent outage with no comms. So disappointed in Supabase.

<img width="450" height="840" alt="Image" src="https://github.com/user-attachments/assets/8171b34a-83d7-4892-9984-d4d493003f0e" />

**Comment by jdgamble555** [comment]
@chanmathew - IDK if they are ever going to update the docs as it has been a few months and they are "deep" into updating them still... I think they are not making this a priority.

Either way, the actual issue is fixed if you implement it correctly.  Here is my SvelteKit version:

https://github.com/supabase/supabase/issues/40985#issuecomment-3679073327

J

**Comment by rossvold** [comment]
> @chanmathew - IDK if they are ever going to update the docs as it has been a few months and they are "deep" into updating them still... I think they are not making this a priority.
> 
> Either way, the actual issue is fixed if you implement it correctly.  Here is my SvelteKit version:
> 
> https://github.com/supabase/supabase/issues/40985#issuecomment-3679073327
> 
> J

Been running basically this config for 1 months and can confirm that it does work. 

I don't understand why they can't just update the 2 pages of documentation for this.

Why spend time fixing this issue, and then completely ignore updating the correlating to the fix 😅😅

**Comment by ChrisChinchilla** [comment]
Hey all, regarding docs. There's a lot more than 2 pages of docs as part of the change, and until this week, the docs team was just me, and I was also on holiday last week. So, some things do take a little longer than I/we hope. But I am back on it this week, and several pages are already updated. Thanks!

**Comment by vehm** [comment]
> Hey all, regarding docs. There's a lot more than 2 pages of docs as part of the change, and until this week, the docs team was just me, and I was also on holiday last week. So, some things do take a little longer than I/we hope. But I am back on it this week, and several pages are already updated. Thanks!

Hoping this is the same sentiment for everyone in this thread, but the frustration expressed here is by no means directed at one person. I appreciate the work everyone has done on this, I just wish that this issue as a _whole_ received attention sooner.

Thank you for your work on this!

**Comment by magick93** [comment]
So, is there a way to turn this annoying message off?

**Comment by jdgamble555** [comment]
@magick93 - problem is not longer turning off the message, but the docs need to be updated so the message won't be seen.  

See: https://github.com/supabase/supabase/issues/40985#issuecomment-3679073327

J

**Comment by mandarini** [comment]
I think we can close this issue now. What do you think @jdgamble555 ?
